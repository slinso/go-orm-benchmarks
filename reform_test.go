package main_test

import (
	"database/sql"
	"testing"

	"github.com/efectn/go-orm-benchmarks/bench"
	"github.com/efectn/go-orm-benchmarks/bench/reform"
	"github.com/gkampitakis/go-snaps/snaps"
	_ "github.com/jackc/pgx/v4/stdlib"
	reformware "gopkg.in/reform.v1"
	"gopkg.in/reform.v1/dialects/postgresql"
)

func TestReform_ReadSlice(t *testing.T) {
	var err error
	db, err := sql.Open("pgx", "host=localhost user=postgres password=postgres dbname=test sslmode=disable")
	if err != nil {
		t.Fatal(err)
	}

	conn := reformware.NewDB(db, postgresql.Dialect, nil)

	m := bench.NewReformModel()
	for i := 0; i < 100; i++ {
		err := conn.Save(m)
		if err != nil {
			t.Fatal(err)
		}
	}

	data, err := conn.SelectAllFrom(reform.ReformModelsTable, "WHERE id > 0 LIMIT 100")
	if err != nil {
		t.Fatal(err)
	}

	snaps.MatchSnapshot(t, data)
}
