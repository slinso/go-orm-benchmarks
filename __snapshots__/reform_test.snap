
[TestReform_ReadSlice - 1]
[]reform.Struct{
    &reform.ReformModels{ID:3, Name:"Orm Benchmark", Title:"Just a Benchmark for fun", Fax:"99909990", Web:"http://blog.milkpod29.me", Age:100, Right:true, Counter:1000},
    &reform.ReformModels{ID:4, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:5, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:6, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:7, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:8, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:9, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:10, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:11, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:12, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:13, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:14, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:15, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:16, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:17, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:2, Name:"Orm Benchmark", Title:"Just a Benchmark for fun", Fax:"99909990", Web:"http://blog.milkpod29.me", Age:100, Right:true, Counter:1000},
    &reform.ReformModels{ID:18, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:19, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:20, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:21, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:22, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:23, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:24, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:25, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:26, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:27, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:28, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:29, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:30, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:31, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:32, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:1, Name:"Orm Benchmark", Title:"Just a Benchmark for fun", Fax:"99909990", Web:"http://blog.milkpod29.me", Age:100, Right:true, Counter:1000},
    &reform.ReformModels{ID:33, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:34, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:35, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:36, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:37, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:38, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:39, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:40, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:41, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:42, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:43, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:44, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:45, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:46, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:47, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:48, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:49, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:50, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:51, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:52, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:53, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:54, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:55, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:56, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:57, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:58, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:59, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:60, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:61, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:62, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:63, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:64, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:65, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:66, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:67, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:68, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:69, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:70, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:71, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:72, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:73, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:74, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:75, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:76, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:77, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:78, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:79, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:80, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:81, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:82, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:83, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:84, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:85, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:86, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:87, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:88, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:89, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:90, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:91, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:92, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:93, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:94, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:95, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:96, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:97, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:98, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:99, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &reform.ReformModels{ID:100, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
}
---
